pkg/core/storage/memcached_store.go

```diff
@@ -90,6 +90,16 @@ func (s *MemCachedStore) GetBatch() *MemBatch {
 
 // Seek implements the Store interface.
 func (s *MemCachedStore) Seek(key []byte, f func(k, v []byte)) {
+	seekres := s.SeekAsync(key, false)
+	for kv := range seekres {
+		f(kv.Key, kv.Value)
+	}
+}
+
+// SeekAsync returns non-buffered channel with matching KeyValue pairs. Key and
+// value slices may not be copied and may be modified. SeekAsync can guarantee
+// that key-value items are sorted by key in ascending way.
+func (s *MemCachedStore) SeekAsync(key []byte, cutPrefix bool) chan KeyValue {
```

Comment:
This interface allows to get the first item and forget, so I think we need some form of iterator with `Close()` now. Otherwise our goroutines are leaking.

Reply:
In theory this also allows us to reuse iterators from persistent storage backends. Refs #2187 .